lib-tracking/src/main/java/ai/kognition/pilecv4j/tracking/tracker/TrackerCSRT.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    static {
+        try {
+            nativeCreate = org.opencv.tracking.TrackerCSRT.class.getDeclaredMethod("create_0");
+            nativeCreate.setAccessible(true);
+            nativeDelete = org.opencv.tracking.TrackerCSRT.class.getDeclaredMethod("delete", long.class);
+            nativeDelete.setAccessible(true);
+        } catch(final NoSuchMethodException | SecurityException e) {
+            throw new RuntimeException("Got an exception trying to access " + TrackerCSRT.class.getSimpleName() +
+                ".delete. Either the security model is too restrictive or the version of OpenCv can't be supported.", e);
+        }
+    }
+
+    public TrackerCSRT() {
+        this(doNativeCreate());
+    }
+
+    protected TrackerCSRT(final long nativeAddr) {
+        super(nativeAddr);
+        stackTrace = TRACK_MEMORY_LEAKS ? new RuntimeException("Here's where I was instantiated: ") : null;
+    }
+
+    @Override
+    public boolean supportsMasking() {
+        return SUPPORTS_MASKING;
+    }
+
+    @Override
+    public Tracker setMask(final Mat mask) {
+        super.setInitialMask(mask);
+        return this;
+    }
+
+    @Override
+    public Optional<Tracker> initialize(final Mat image, final Rect2d initialBoundingBox) {
+        if(!super.init(image, initialBoundingBox)) {
+            this.close();
+            return Optional.empty();
+        }
+        isInitialized = true;
+        return Optional.of(this);
+    }
+
+    @Override
+    public boolean isInitialized() {
+        return isInitialized;
+    }
+
+    @Override
+    public Optional<Rect2d> update(final Mat image) {
+        final Rect2d retval = new Rect2d();
+        if(!super.update(image, retval))
+            return Optional.empty();
+        return Optional.of(retval);
+    }
+
+    protected static long doNativeCreate() {
+        try {
+            return (Long)nativeCreate.invoke(null);
+        } catch(final IllegalAccessException | InvocationTargetException e) {
+            throw new RuntimeException("Got an exception trying to call Tracker.create_0. Either the security model is too restrictive or the version of " +
+                "OpenCV can't be supported.", e);
+        }
+    }
+
+    protected void doNativeDelete() {
+        try {
+            nativeDelete.invoke(this, super.nativeObj);
+        } catch(final IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+            throw new RuntimeException("Got an exception trying to call Tracker.delete. Either the security model is too restrictive or the version of " +
+                "OpenCV can't be supported.", e);
+        }
+    }
+
+    @Override
+    public TrackerCSRT skipOnceForReturn() {
+        skipOnceForDelete = true;
+        return this;
+    }
+
+    @Override
+    public void close() {
+        if(!skipOnceForDelete) {
+            if(!deletedAlready) {
+                doNativeDelete();
+                deletedAlready = true;
```

Comment:
So there are two easy options here:

1. Add the javax annotation `@NotThreadSafe` (see 2)
2. From the `@NotThreadSafe` annotation: 
> NotThreadSafe The class to which this annotation is applied is not thread-safe. This annotation primarily exists for clarifying the non-thread-safety of a class that might otherwise be assumed to be thread-safe, despite the fact that it is a bad idea to assume a class is thread-safe without good reason.

I'm in favor of doing nothing not because it's the easiest for me, but 
1. There's not a huge amount of reason to expect that people would use a time-ordered-sensitive tracker in a multi-threaded context. If they are, I feel like that's beyond the intentions of this class.
2. It's dangerous to assume things that don't explicitly make reference to thread safety are thread-safe

Reply:
Ok